assembl/static2/js/app/utils/httpRequestHandler.js

```diff
@@ -43,22 +43,15 @@ export const xmlHttpRequest = (obj) => {
         }
         resolve(resp);
       } else {
-        let resp;
-        // TODO: Agree on contract with backend for all APIs!!
-        // Current system not very intuitive nor friendly
-        try {
-          resp = JSON.parse(xhr.responseText);
-          reject(resp || xhr.statusText);
-        } catch (e) {
-          // A non-JSONError response, likely from Pyramid itself
-          // Short term solution until contract established
-          reject(xhr.status);
-        }
+        // Contract agreed upon. If API is to fail, must respond with
+        // JSONError type. Front-end respects this type of response only.
```

Comment:
I would not check status range at all. Use the mimetype to know if it's a jsonError, now that the mimetype is correct.